apps/web/pages/index.vue

```diff
@@ -1,48 +1,92 @@
 <template>
-  <Editor v-if="isEditing" />
-  <div v-else class="content">
-    <div class="max-w-screen-3xl mx-auto md:px-6 lg:px-10 mb-10">
-      <UiHeroCarousel :hero-item-props="homepage.hero" />
-    </div>
-
-    <NuxtLazyHydrate when-visible>
-      <div class="max-w-screen-3xl mx-auto md:px-6 lg:px-10 mb-10">
-        <UiMediaCard
-          v-for="(item, index) in homepage.mediaCard"
-          :key="index"
-          :image="item.image"
-          :alt="item.alt"
-          :text="item.text"
-          :alignment="item.alignment"
-        />
-      </div>
-    </NuxtLazyHydrate>
-
-    <div class="max-w-screen-3xl mx-auto md:px-6 lg:px-10 mb-10">
-      <template v-for="(item, index) in homepage.featured" :key="index">
-        <section class="mb-10 overflow-hidden">
-          <p data-testid="recommended-products" class="mb-4 typography-text-lg text-center md:text-left">
-            {{ item.headline }}
-          </p>
-          <ProductRecommendedProducts cache-key="homepage" :category-id="item.categoryId" />
-        </section>
+  <div>
+    <Editor
+      v-if="isEditing && currentBlockIndex !== null"
+      :index="currentBlockIndex"
+      :block="currentBlock"
```

Comment:
**suggestion**: Only pass the block index and have the editor retrieve the correct block based on that. This way we always work with and modify the same data source.

Reply:
Will be addressed separately.